nautobot/docs/models/dcim/deviceredundancygroups.md

```diff
@@ -0,0 +1,250 @@
+    devices(name__ie: $device_name) {
+        name
+        device_redundancy_group {
+            name
+            members {
+                name
+                device_redundancy_group_priority
+                interfaces(name__ie: "failover-link") {
+                    type
+                    name
+                    ip_addresses {
+                        host
+                        prefix_length
+                    }
+                    parent_interface {
+                        name
+                        type
+                    }
+                }
+            }
+        }
+    }
+}
+"""
+
+def get_gql_failover_details(device_name):
+    variables = {"device_name": device_name}
+    nb = pynautobot.api(
+        url="http://localhost:8080",
+        token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+    )
+    return nb.graphql.query(query=query, variables=variables)
+```
+
+### Retrieving the data - Primary Failover Unit ("nyc-fw-primary")
+
+An example data returned from Nautobot is presented below.
+
+```python
+>>> primary_unit_details = get_gql_failover_details("nyc-fw-primary").json
+```
+
+```json
+{'data': {'devices': [{'name': 'nyc-fw-primary', 'device_redundancy_group': {'name': 'nyc-firewalls', 'members': [{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.0', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}, {'name': 'nyc-fw-secondary', 'device_redundancy_group_priority': 50, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.1', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}]}}]}}
+```
+
+### Retrieving the data - Secondary Failover Unit ("nyc-fw-secondary")
+
+An example data returned from Nautobot is presented below.
+
+```python
+>>> secondary_unit_details = get_gql_failover_details("nyc-fw-secondary").json
+```
+
+```json
+{'data': {'devices': [{'name': 'nyc-fw-secondary', 'device_redundancy_group': {'name': 'nyc-firewalls', 'members': [{'name': 'nyc-fw-primary', 'device_redundancy_group_priority': 100, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.0', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}, {'name': 'nyc-fw-secondary', 'device_redundancy_group_priority': 50, 'interfaces': [{'type': 'VIRTUAL', 'name': 'failover-link', 'ip_addresses': [{'host': '172.27.48.1', 'prefix_length': 31}], 'parent_interface': {'name': 'gigabitethernet0/3', 'type': 'A_1000BASE_T'}}]}]}}]}}
+```
+
+### Processing the data - Primary Failover Unit ("nyc-fw-primary")
+
+Before rendering the data, we will process the data to simplify data variables within template:
+
+```python
+>>> hostname = primary_unit_details['data']['devices'][0]['name']
```

Comment:
we already knew `hostname` was `nyc-fw-primary` as it was an input to the query, we shouldn't need to re-derive it?